wdl/transforms/new-base/src/main/scala/wdl/transforms/base/ast2wdlom/AstToCommandSectionElement.scala

```diff
@@ -51,24 +75,29 @@ object AstToCommandSectionElement {
     CommandSectionLine(line.parts.filterNot(empty))
   }
 
-  private def minimumStartLength(lines: Vector[CommandSectionLine]): Int = {
+  private def leadingWhitespace(lines: Vector[CommandSectionLine]): Vector[String] = {
     val r = "^(\\s*)".r
-    def startLength(line: CommandSectionLine) = line.parts.headOption match {
-      case Some(StringCommandPartElement(str)) => r.findFirstIn(str).map(_.length).getOrElse(0)
-      case _ => 0
+
+    def leadingWhitespaceForLine(line: CommandSectionLine): Option[String] = line.parts.headOption match {
+      case Some(StringCommandPartElement(str)) => r.findFirstIn(str)
+      case _ => None
     }
 
-    if (lines.nonEmpty) lines.map(startLength).min else 0
+    lines.map(leadingWhitespaceForLine(_).getOrElse(""))
   }
 
-  private def stripStarts(lines: Vector[CommandSectionLine], prefix: String): Vector[CommandSectionLine] = {
-    if (prefix.isEmpty) lines else lines map { line =>
-      line.parts.headOption match {
-        case Some(StringCommandPartElement(str)) if str.startsWith(prefix) =>
-          CommandSectionLine(Vector(StringCommandPartElement(str.stripPrefix(prefix))) ++ line.parts.tail)
-        case _ => ??? // TODO: Should be impossible. Can we make the function total by fixing the inputs?
-      }
-    }
+  private def stripStarts(lines: Vector[CommandSectionLine], prefix: String): Checked[List[CommandSectionLine]] = {
+    if (prefix.isEmpty)
+      lines.toList.validNelCheck
+    else
+      (lines map { line =>
+        line.parts.headOption match {
+          case Some(StringCommandPartElement(str)) if str.startsWith(prefix) =>
+            CommandSectionLine(Vector(StringCommandPartElement(str.stripPrefix(prefix))) ++ line.parts.tail).validNelCheck
+          case _ =>
+            "Failed to strip common whitespace prefix from line.".invalidNelCheck
+        }
+      }).toList.sequence[Checked, CommandSectionLine]
```

Comment:
I think `map` + `sequence` = `traverse`, but don't worry too much (especially if intellij has trouble picking up the types)